contrib/Shopify/sarama/option.go

```diff
@@ -17,3 +19,19 @@ func WithServiceName(name string) Option {
 		cfg.serviceName = name
 	}
 }
+
+// WithAnalytics enables Trace Analytics for all started spans.
+func WithAnalytics(on bool) Option {
```

Comment:
Good question. I didn't want to have a boolean initially, but I figured it's better to have it in case of:
* The default state will become _enabled_ and users will want to disable it.
* The global configuration enables analytics on everything, but you want to disable it on some integrations.

Reply:
Correct. The fact that we have the APM events sample rate, is an edge case and we don't expect people to use it. That said, it's still available to cover edge cases. Let's keep the boolean.